keps/sig-node/4569-cgroup-v1-deprecation/README.md

```diff
@@ -0,0 +1,685 @@
+
+<!--
+In principle every added code should have complete unit test coverage, so providing
+the exact set of tests will not bring additional value.
+However, if complete unit test coverage is not possible, explain the reason of it
+together with explanation why this is acceptable.
+-->
+
+<!--
+Additionally, for Alpha try to enumerate the core package you will be touching
+to implement this enhancement and provide the current unit coverage for those
+in the form of:
+- <package>: <date> - <current test coverage>
+The data can be easily read from:
+https://testgrid.k8s.io/sig-testing-canaries#ci-kubernetes-coverage-unit
+This can inform certain test coverage improvements that we want to do before
+extending the production code to implement this enhancement.
+-->
+
+The respective kubelet subcomponents already have unit tests cases to handle cgroup v2. However, additional unit tests will be added to test the feature flag `DeprecatedCgroupV1` and the kubelet flag `--suppress-cgroupv1-deprecation-warning` combinations.
+
+##### Integration tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+-->
+
+N/A
+
+##### e2e tests
+
+<!--
+This question should be filled when targeting a release.
+For Alpha, describe what tests will be added to ensure proper quality of the enhancement.
+For Beta and GA, add links to added tests together with links to k8s-triage for those tests:
+https://storage.googleapis.com/k8s-triage/index.html
+We expect no non-infra related flakes in the last month as a GA graduation criteria.
+-->
+
+1. Monitor existing cgroup v2 jobs, e.g.
+  - https://testgrid.k8s.io/sig-node-release-blocking#ci-crio-cgroupv2-node-e2e-conformance
+
+2. Migrate existing e2e and node e2e jobs that still use cgroup v1 to cgroup v2.
+
+### Graduation Criteria
+
+#### Alpha
+
+- Feature gate `DeprecatedCgroupV1` is introduced, disabled by default.
+- When the `DeprecatedCgroupV1` feature gate is enabled and the kubelet detects the host using cgroup v1, it will not only log a warning message but also generate an event to highlight the deprecation of cgroup v1.
+- To minimize log verbosity and event clutter resulting from the `DeprecatedCgroupV1` feature being enabled on hosts still using cgroup v1, a new kubelet flag, `--suppress-cgroupv1-deprecation-warning`, is introduced.
+
+#### Beta
+
+- `DeprecatedCgroupV1` feature gate is enabled by default.
+- Fail the kubelet on start up if the host is still using cgroup v1.
+- Retain `--suppress-cgroupv1-deprecation-warning` flag introduced in `alpha`.
+- Migrate existing e2e and node e2e jobs that still use cgroup v1 to cgroup v2.
+
+#### GA
+
+- Remove feature flag `DeprecatedCgroupV1`.
+- Remove kubelet flag `--suppress-cgroupv1-deprecation-warning`.
```

Comment:
we can not add and remove flags like this, flags are apis for users and this can break installers